Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,23 +98,42 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
-            let parsedAttributes = try parseAttributes(
-                name: name,
-                platform: platform,
-                requiredTemplateOptions: requiredTemplateOptions,
-                optionalTemplateOptions: optionalTemplateOptions,
-                template: template
-            )
+            var template: Template?
```

Comment:
```suggestion
            let template: Template
```

You can do this if you refactor the redundant code from the `if-else`